From 23a676ad606ddd453cb204752861e1e8bebea8c9 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Fri, 18 Sep 2026 20:47:14 +0100
Subject: [PATCH 34/36] replace removed shell_devpts type with devpts in su policy

aosp dropped the shell_devpts type; it existed only as a compatibility
alias for older platform API levels, and is gone entirely by the 202404
api level.  the gsi system_ext policy still referenced it in su.te,
which fails the checkpolicy pass for system_ext_sepolicy.cil with
"unknown type shell_devpts".

devpts is the current umbrella type for pts files and is what aosp's own
shell.te and adbd_common.te use for the equivalent rules.
---
 sepolicy/su.te | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sepolicy/su.te b/sepolicy/su.te
index ae430a6..9650e5e 100644
--- a/sepolicy/su.te
+++ b/sepolicy/su.te
@@ -35,7 +35,7 @@ allow system_server phhsu_daemon:fifo_file { read write getattr };
 allow system_server phhsu_daemon:fd use;
 allow system_server phhsu_daemon:binder { call transfer };
 allow system_server phhsu_daemon_tmpfs:file { getattr read open };
-allow system_server shell_devpts:chr_file { read write };
+allow system_server devpts:chr_file { read write };
 
 # Add su to various domains
 net_domain(phhsu_daemon)
@@ -72,7 +72,7 @@ allow { phhsu_daemon shell } rootfs:file { rwx_file_perms create rename setattr
 allow { phhsu_daemon shell } rootfs:dir rw_dir_perms;
 allow phhsu_daemon asec_apk_file:dir rw_dir_perms;
 
-allow phhsu_daemon shell_devpts:chr_file rw_file_perms;
+allow phhsu_daemon devpts:chr_file rw_file_perms;
 
 allow phhsu_daemon app_data_file:dir rw_dir_perms;
 allow phhsu_daemon app_data_file:file { rwx_file_perms create rename setattr unlink };
-- 
2.55.0

